### PR TITLE
Migrate utility scripts to separate repository

### PR DESCRIPTION
**Pull request by sbesson**
In addition to the curated study metadata, this repository contains a few utility scripts used routinely in the IDR, notably

- `stats.py` used for computing statistics for each IDR release
- `study_parser.py` for parsing and validating study files and annotating the resource
- various utility scripts for (re-)annotating studies with bulk annotations and map annotations

The primary issue with the current layout is the size of the idr-metadata repository which increase the cost of modifying these scripts for new features, bug fixes.

This PR proposes to migrate away all utility scripts to a dedicated `idr-utils` repository. The repository is still included as a submodule so that the Travis checks can be executed. The `mini-idr` repository is also removed as we have more robust mechanisms to deploy complete copies of IDR for testing software via IDR/deployment.

Eventually the Python based utility could be made easier to deploy by making them pip-installable and maybe reconciling them with `idr-py`. Splitting them into their own repository is also a good first step to achieving this goal.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,19 @@
...
-# http://docs.travis-ci.com/user/workers/container-based-infrastructure/
+python: 2.7
 sudo: false
 
 cache: pip
 
 install:
-  - pip install PyYAML "pandas<0.19" "flake8<3.6.0" yamllint
-  - python setup.py install
+  - pip install -r requirements.txt
```
Comment (joshmoore) [inline .travis.yml]: Could this be in the setup.py of idr-utils?

File: .travis.yml [hunk 2]
```diff
@@ -1,26 +1,19 @@
...
+  - pip install idr-utils/
```
Comment (joshmoore) [inline .travis.yml]: Interesting. What does this do?

File: .travis.yml (hunk 1 above)
Comment (sbesson) [reply .travis.yml]: At the moment, they are in `idr-utils/requirements.txt` (with `flake8` being uncapped). Can have them in `idr-utils/setup.py` instead. Eventually we should be able to install all requirements as depedency of the utils library.

File: .travis.yml (hunk 2 above)
Comment (sbesson) [reply .travis.yml]: I think largely equivalent to `cd idr-utils && python setup.py install`. I remember there are cases where this form was preferable but I would need to dig up the threads.

**Comment by sbesson** [comment]
No particular reason for keeping `stats.py` top-level, I can migrate it under `idr-utils/scripts` (and push the submodule to the org if you are generally happy with the concept.

File: .travis.yml (hunk 2 above)
Comment (joshmoore) [reply .travis.yml]: Agreed, but does it _do_ anything? i.e. if we're always using `python idr-utils/...` what's the purpose?

Comment (sbesson) [reply .travis.yml]: Trying to remember, I might have been overoptimistic and expecting `pip install idr-utils/` to work out-of-the box  (requirements, scripts in the path). Getting it working properly obviously needs a bit more work :) I'll push it away

File: .travis.yml (hunk 1 above)
Comment (sbesson) [reply .travis.yml]: 961701b pushed to install the requirements from `idr-utils`

File: .travis.yml [hunk 3]
```diff
@@ -5,9 +5,7 @@ sudo: false
 
 cache: pip
 
-install:
-  - pip install -r requirements.txt
```
Comment (joshmoore) [inline .travis.yml]: Does this mean that requirements.txt can be removed?

Comment (sbesson) [reply .travis.yml]: :+1: 4ab30a8

**Pull request merged**